### PR TITLE
Jiochat is a sender

### PR DESCRIPTION
**Pull request by norkans7**

### CODE REVIEW

File: temba/channels/tests.py [hunk 1]
```diff
@@ -2121,7 +2121,7 @@ def test_claim_twitter_beta(self, mock_verify_credentials, mock_register_webhook
         self.assertEqual(json.loads(channel.config), {'handle_id': '87654', 'api_key': 'ak', 'api_secret': 'as',
                                                       'access_token': 'at', 'access_token_secret': 'ats'})
 
-        mock_register_webhook.assert_called_once_with('https://temba.ngrok.io/handlers/twitter/%s' % channel.uuid)
+        mock_register_webhook.assert_called_once_with('https://%s%s' % (settings.HOSTNAME, reverse('handlers.twitter_handler', args=[channel.uuid])))
```
Comment (norkans7) [inline temba/channels/tests.py]: Just use the settings variable

File: temba/channels/views.py [hunk 2]
```diff
@@ -534,19 +534,10 @@ def channel_status_processor(request):
...
-
-        # and facebook
-        if not send_channel:
-            send_channel = org.get_send_channel(scheme=FACEBOOK_SCHEME)
-
-        # and viber
-        if not send_channel:
-            send_channel = org.get_send_channel(scheme=VIBER_SCHEME)
+            for scheme in SEND_URN_SCHEMES:
+                send_channel = org.get_send_channel(scheme=scheme)
```
Comment (rowanseymour) [inline temba/channels/views.py]: What are we trying to do here? If this is just about determining whether we have a channel that can send, then it seems more efficient to allow `get_send_channel` to be called without a specific scheme.

Comment (norkans7) [inline temba/channels/views.py]: We need to specify either a scheme or a contact URN 

https://github.com/nyaruka/rapidpro/blob/master/temba/orgs/models.py#L458

Comment (rowanseymour) [inline temba/channels/views.py]: Sure but we can change that. What I'm getting at here is that `get_send_channel` is used in different ways - when we're creating an outgoing message we use it to find a channel that can send to a given URN. But it looks like we might be using it here to determine "are there any send channels" - and we answer that question in a more efficient way than looping through each scheme looking for a channel.

File: temba/orgs/models.py [hunk 3]
```diff
@@ -433,16 +433,24 @@ def get_channel(self, scheme, country_code, role):
...
         channel = channel.first()
 
         # no channel? try without country
         if not channel and country_code:
-            channel = self.channels.filter(is_active=True, scheme=scheme,
-                                           role__contains=role).order_by('-pk').first()
+            channel = self.channels.filter(is_active=True, role__contains=role).order_by('-pk')
+
+            if scheme is not None:
+                channel = channel.filter(scheme=scheme)
```
Comment (rowanseymour) [inline temba/orgs/models.py]: Don't need to repeat lines 436, 438, 439. Change those so they are assigning to a var called `channels`, and then you can reuse that var here

File: temba/channels/views.py [hunk 4]
```diff
@@ -505,19 +505,7 @@ def channel_status_processor(request):
...
-            send_channel = org.get_send_channel(scheme=TELEGRAM_SCHEME)
-
-        # and facebook
-        if not send_channel:
-            send_channel = org.get_send_channel(scheme=FACEBOOK_SCHEME)
-
-        # and viber
-        if not send_channel:
-            send_channel = org.get_send_channel(scheme=VIBER_SCHEME)
+            send_channel = org.get_send_channel()
```
Comment (rowanseymour) [inline temba/channels/views.py]: Do we still need the call on line 502? (I haven't checked how this is used so not sure)

**Pull request merged**